spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/influx/InfluxDbAutoConfiguration.java

```diff
@@ -53,8 +60,11 @@ public InfluxDB influxDb() {
 		if (Strings.isNullOrEmpty(this.properties.getUser())) {
 			return InfluxDBFactory.connect(this.properties.getUrl());
 		}
+		if (this.builder != null) {
+			return InfluxDBFactory.connect(this.properties.getUrl(),
+					this.properties.getUser(), this.properties.getUrl(), this.builder);
```

Comment:
There is no check on user/password there so perhaps we could rationalize the check altogether? (i.e. remove the first if)